src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -31,113 +29,214 @@ public OverridingAssignmentAnalysis(Body methodBody, AbstractMergeConflictDefini
-            String localName = getLocalName((Local) v.getValue());
-            for (DataFlowAbstraction filterIn: in){
-                String inName = getLocalName(filterIn.getLocal());
+        getStatementAssociatedWithUnit(u).forEach(valueBox -> {
+            String localName = getLocalValueBoxName(valueBox);
+
+            for (DataFlowAbstraction filterIn : in) {
+                String inName = getLocalDataFlowAbstraction(filterIn);
 
-                if (filterIn.getStmt().getType().equals(Statement.Type.SOURCE) && inName.equals(localName)){
+                if (filterIn.getStmt().getType().equals(Statement.Type.SOURCE) && inName.equals(localName)) {
                     left.add(filterIn);
-                }else if (filterIn.getStmt().getType().equals(Statement.Type.SINK) && inName.equals(localName)){
+                } else if (filterIn.getStmt().getType().equals(Statement.Type.SINK) && inName.equals(localName)) {
                     right.add(filterIn);
                 }
             }
         });
 
-        if(isSinkStatement(u)) {
+        if (isRightStatement(u)) {
             checkConflicts(u, left);
-        }else  if (isSourceStatement(u)){
+        } else if (isLeftStatement(u)) {
             checkConflicts(u, right);
         }
     }
 
-    protected void checkConflicts(Unit u, List<DataFlowAbstraction> statements){
-        for (DataFlowAbstraction statement : statements) {
-            if (statementEquals(statement, u)) {
-                Conflict c = new Conflict(statement.getStmt(), findStatement(u));
+    /*
+     * Checks if there is a conflict and if so adds it to the conflict list.
+     */
+    private void checkConflicts(Unit u, List<DataFlowAbstraction> dataFlowAbstractionList) {
+        for (DataFlowAbstraction dataFlowAbstraction : dataFlowAbstractionList) {
+            if (checkEqualVariablesName(dataFlowAbstraction, u)) {
+                Conflict c = new Conflict(dataFlowAbstraction.getStmt(), findStatement(u));
                 Collector.instance().addConflict(c);
             }
         }
     }
+    /*
+     * Checks the equality of variable names between an
+     * item in the abstraction list and all the UnitBox's ValueBox.
+     */
+    private boolean checkEqualVariablesName(DataFlowAbstraction dataFlowAbstraction, Unit u) {
+        for (ValueBox valueBox : u.getDefBoxes()) {
+            return getLocalDataFlowAbstraction(dataFlowAbstraction).equals(getLocalValueBoxName(valueBox));
+        }
+        return false;
+    }
 
-    @Override
-    protected void flowThrough(FlowSet<DataFlowAbstraction> in, Unit u, FlowSet<DataFlowAbstraction> out) {
-        detectConflict(in, u);
-        FlowSet<DataFlowAbstraction> temp = new ArraySparseSet<>();
+    /*
+     * Returns a String containing the name of the variable given a DataFlowAbstraction
+     */
+    private String getLocalDataFlowAbstraction(DataFlowAbstraction dataFlowAbstraction) {
```

Comment:
@barbosamaatheus wouldn't gerVarNameInAbstraction be better? it seems to match better the comment